### ISSUE TITLE
NPE when trying access to Jenkins API with auth view github

### ISSUE THREAD

**Issue by Prototik** [mocleiri/github-oauth-plugin#36]
Hi. I need to use Jenkins API (JSON), but when i'm trying use with account on github - NPE and HTTP 500 error.

```
INFO   | jvm 1    | 2013/01/19 13:00:09 | WARNING: Untrapped Error in Servlet
INFO   | jvm 1    | 2013/01/19 13:00:09 | java.lang.NullPointerException
INFO   | jvm 1    | 2013/01/19 13:00:09 |   at org.jenkinsci.plugins.GithubSecurityRealm.loadGroupByGroupname(GithubSecurityRealm.java:459)
INFO   | jvm 1    | 2013/01/19 13:00:09 |   at org.jenkinsci.plugins.GithubSecurityRealm.loadUserByUsername(GithubSecurityRealm.java:428)
INFO   | jvm 1    | 2013/01/19 13:00:09 |   at hudson.model.User.impersonate(User.java:254)
INFO   | jvm 1    | 2013/01/19 13:00:09 |   at jenkins.security.ApiTokenFilter.doFilter(ApiTokenFilter.java:52)
INFO   | jvm 1    | 2013/01/19 13:00:09 |   at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)
INFO   | jvm 1    | 2013/01/19 13:00:09 |   at org.acegisecurity.context.HttpSessionContextIntegrationFilter.doFilter(HttpSessionContextIntegrationFilter.java:249)
INFO   | jvm 1    | 2013/01/19 13:00:09 |   at hudson.security.HttpSessionContextIntegrationFilter2.doFilter(HttpSessionContextIntegrationFilter2.java:66)
INFO   | jvm 1    | 2013/01/19 13:00:09 |   at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)
INFO   | jvm 1    | 2013/01/19 13:00:09 |   at hudson.security.ChainedServletFilter.doFilter(ChainedServletFilter.java:76)
INFO   | jvm 1    | 2013/01/19 13:00:09 |   at hudson.security.HudsonFilter.doFilter(HudsonFilter.java:164)
INFO   | jvm 1    | 2013/01/19 13:00:09 |   at winstone.FilterConfiguration.execute(FilterConfiguration.java:194)
INFO   | jvm 1    | 2013/01/19 13:00:09 |   at winstone.RequestDispatcher.doFilter(RequestDispatcher.java:366)
INFO   | jvm 1    | 2013/01/19 13:00:09 |   at org.kohsuke.stapler.compression.CompressionFilter.doFilter(CompressionFilter.java:50)
INFO   | jvm 1    | 2013/01/19 13:00:09 |   at winstone.FilterConfiguration.execute(FilterConfiguration.java:194)
INFO   | jvm 1    | 2013/01/19 13:00:09 |   at winstone.RequestDispatcher.doFilter(RequestDispatcher.java:366)
INFO   | jvm 1    | 2013/01/19 13:00:09 |   at hudson.util.CharacterEncodingFilter.doFilter(CharacterEncodingFilter.java:81)
INFO   | jvm 1    | 2013/01/19 13:00:09 |   at winstone.FilterConfiguration.execute(FilterConfiguration.java:194)
INFO   | jvm 1    | 2013/01/19 13:00:09 |   at winstone.RequestDispatcher.doFilter(RequestDispatcher.java:366)
INFO   | jvm 1    | 2013/01/19 13:00:09 |   at winstone.RequestDispatcher.forward(RequestDispatcher.java:331)
INFO   | jvm 1    | 2013/01/19 13:00:09 |   at winstone.RequestHandlerThread.processRequest(RequestHandlerThread.java:215)
INFO   | jvm 1    | 2013/01/19 13:00:09 |   at winstone.RequestHandlerThread.run(RequestHandlerThread.java:138)
INFO   | jvm 1    | 2013/01/19 13:00:09 |   at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471)
INFO   | jvm 1    | 2013/01/19 13:00:09 |   at java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:334)
INFO   | jvm 1    | 2013/01/19 13:00:09 |   at java.util.concurrent.FutureTask.run(FutureTask.java:166)
INFO   | jvm 1    | 2013/01/19 13:00:09 |   at winstone.BoundedExecutorService$1.run(BoundedExecutorService.java:77)
INFO   | jvm 1    | 2013/01/19 13:00:09 |   at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1110)
INFO   | jvm 1    | 2013/01/19 13:00:09 |   at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:603)
INFO   | jvm 1    | 2013/01/19 13:00:09 |   at java.lang.Thread.run(Thread.java:722)
```

If i use incorrect user id or token - 401, all right. But when login and password correct - 500.

**Comment by rgriffith** [comment]
Just wanted to add that I receive the same NPE when attempting to use the [Hubot Jenkins script](https://github.com/github/hubot-scripts/blob/master/src/scripts/jenkins.coffee) with our Jenkins instance that has the Github auth plugin.

**Comment by ghost** [comment]
Sounds like issue #18 

A fix was accepted to master months ago, but a new release hasn't been made. Your choices are to abandon GitHub auth or build/install the plugin manually.

**Comment by rgriffith** [comment]
Thanks for the update @toolbear. Confirmed the fix for issue #18 does fix the error and allowed me to use the Hubot script along with the auth plugin.

**Comment by mocleiri** [comment]
I'm no longer maintaining the plugin.  

@skottler is but using the main repository here: https://github.com/jenkinsci/github-oauth-plugin

Normally jenkins projects use jira for the issues but I never turned them off here.  But I think its confusing for reporters to report here. 

I want to turn off the issues on this repository so can you refile the ticket into the [jenkins jira for the _github-oauth_ component](https://issues.jenkins-ci.org/secure/IssueNavigator.jspa?reset=true&jqlQuery=project+%3D+JENKINS+AND+status+in+%28Open%2C+%22In+Progress%22%2C+Reopened%29+AND+component+%3D+%27github-oauth%27)?

Unfortunately there doesn't seem to be a way to make the existing issues _read-only_ so I can disable issues without all of the existing reported and resolved issues vanishing.

Thanks
